Descriptor[] descriptors
---
float64[] parts_classes
